miniml/compiler/test/functions.ml

```diff
@@ -43,6 +43,10 @@ let () =
 let () = print_int (let a = 17 in let b = 42 in if (let x = 2 in true) then a else b)
 let () = print_int (let a = 17 in let b = 42 in if (let x = 2 in false) then a else b)
 
+(* this ensures that the 'let' are not substituted away... as long as we don't support constant-folding *)
+let () = print_int (let a = 16+1 in let b = 41+1 in if (let x = 1+1 in true) then a else b)
+let () = print_int (let a = 16+1 in let b = 41+1 in if (let x = 1+1 in false) then a else b)
```

Comment:
I think the same change should be done in `exits.ml` for the test where a local variable is used to avoid having the exception handler as the top of the stack.

Reply:
Ideally we would have the equivalent of `Sys.opaque_identity`, but I am slightly too lazy to implement that.